lib/msf/core/data_store.rb

```diff
@@ -216,6 +219,36 @@ def from_file(path, name = 'global')
     end
   end
 
+  #
+  # Return a deep copy of this datastore.
+  #
+  def copy
+    clone = self.class.new
+    self.keys.each do |k|
+      clone.import_option(k, self[k].kind_of?(String) ? self[k].dup : self[k], @imported[k], @imported_by[k])
+    end
+    clone.aliases = self.aliases.dup
+    clone
+  end
+
+  #
+  # Override merge! so that we merge the aliases and imported hashes
+  #
+  def merge!(other)
+    super
+    self.aliases.merge!(other.aliases) if other.respond_to?(:aliases)
```

Comment:
This works, alternately you can do:
```ruby
if other.is_a? DataStore
  self.aliases.merge!(other.aliases)
  .....
end
```
Which seems a bit cleaner to me, but that's up to you.

Reply:
Since ModuleDataStore is_a DataStore, sure.